server/src/services/frequency.py

```diff
@@ -1,133 +1,186 @@
-        for only the array of neighborhood council #s
-        Returns summary data as well if aggregate is set to True
-        Returns only entries created between startdate and enddate if values
-        are set for those parameters
-        Format of startdate and enddate should be a string in
-        the form 2019-12-01 23:02:05
-        """
-        engine = db.create_engine(self.dbString)
-
-        if service:
-            df = pd.read_sql_query("SELECT\
-            requesttype,\
-            createddate,\
-            closeddate,\
-            servicedate,\
-            nc,\
-            ncname\
-            FROM %s" % self.table, con=engine)
-            df['servicedate'] = pd.to_datetime(df['servicedate'])
+        filteredData = self.dataAccess.query(fields, filters)
+        df = pd.DataFrame(data=filteredData['data'])
 
+        if window == 'month':
+            numBins = 10
         else:
-            df = pd.read_sql_query("SELECT\
-            requesttype,\
-            createddate,\
-            closeddate,\
-            nc,\
-            ncname\
-            FROM %s" % self.table, con=engine)
-
-        df['closeddate'] = pd.to_datetime(df['closeddate'])
-
-        if councils != []:
-            df = df[df.nc.isin(councils)]
-
-        if startdate != "":
-            start = pd.to_datetime(startdate)
-            df = df[(df['createddate'] >= start)]
-
-        if enddate != "":
-            end = pd.to_datetime(enddate)
-            df = df[df['createddate'] <= end]
-
-        df = df.sort_values(by=['requesttype',
-                                'nc',
-                                'createddate',
-                                'closeddate'])
-        df_json = json.loads(df.to_json(orient="records"))
-
-        if aggregate:
-            summary = self.freq_aggregate(df)
-            json_data = []
-            json_data.append(json.loads(summary))
-            json_data.append(df_json)
-            return json_data
-
-        return df_json
-
-# Todo: filter by NC at the sql request stage instead of afterwards
-
-
-if __name__ == "__main__":
-    freq = frequency()
-    config = ConfigParser()
-    config.read("../setting.cfg")
-    freq.config = config
-    freq.dbString = config['Database']['DB_CONNECTION_STRING']
-    freq.freq_view_data(service=True, aggregate=True)
+            numBins = 12
+
+        df['buckets'] = pd.qcut(df['createddate'], q=numBins, precision=0)
+        bucketStartDates = [str(df['buckets'].unique()[i].left)
+                            for i in range(numBins)]
+
+        return [{
+            'bucketStartDates': bucketStartDates,
+            'requestTypes': [{
+                'type': request,
+                'numRequests': df['buckets'][df['requesttype'] == request]
+                .value_counts(sort=False).values.tolist()
+            } for request in requestTypes]
+        }]
+
+# Following is deprecated, saving for reference
```

Comment:
We can remove all this commented stuff
If we need to reference it we can go through git-history. Otherwise this will make the code messy